### PR TITLE
Adheres to Effective Dart

### PR DESCRIPTION
**Pull request by tenhobi**
## Status
**READY**

## Breaking Changes
NO

## Description
Hi, I have added `package:effective_dart` to the project, so we can have some lints. Hope you like it, feel free to add any comment. ❤️

### CODE REVIEW

File: test/custom_list.dart [hunk 1]
```diff
@@ -7,10 +7,12 @@ import 'dart:math';
...
 class CustomList<E> implements List<E> {
   bool _copyBeforeWrite;
-  bool _growable;
+  final bool _growable;
   List<E> _list;
 
-  CustomList(this._list, this._growable) : _copyBeforeWrite = true;
+  CustomList(this._list, {growable = false})
+      : _copyBeforeWrite = true,
+        _growable = growable;
```
Comment (tenhobi) [inline test/custom_list.dart]: Because of `dart(avoid_positional_boolean_parameters)`

File: test/equatable_test.dart [hunk 2]
```diff
@@ -716,28 +717,36 @@ void main() {
 
     group('Set Equatable', () {
       test('should return when values are same', () {
+        // ignore: prefer_collection_literals
         final instanceA = SimpleEquatable<Set>(Set.from(["A", "B"]));
```
Comment (tenhobi) [inline test/equatable_test.dart]: Can't use `{}` for a set, that comes in v2.2 or something.

File: README.md [hunk 3]
```diff
@@ -13,6 +13,9 @@
     <img alt="Pub Package" src="https://img.shields.io/pub/v/equatable.svg">
   </a>
   <br/>
+  <a href="https://github.com/tenhobi/effective_dart">
```
Comment (felangel) [inline README.md]: 💯

File: lib/src/equatable_utils.dart [hunk 4]
```diff
@@ -1,10 +1,12 @@
 import 'package:collection/collection.dart';
 
+/// Retuns a hash code for [props].
```
Comment (felangel) [inline lib/src/equatable_utils.dart]:
```suggestion
/// Returns a `hashCode` for [props].
```

File: lib/src/equatable_utils.dart [hunk 5]
```diff
@@ -1,10 +1,12 @@
...
 int mapPropsToHashCode(Iterable props) =>
-    _finish(props?.fold(0, (hash, object) => _combine(hash, object)) ?? 0);
+    _finish(props?.fold(0, _combine) ?? 0);
 
 const DeepCollectionEquality _equality = DeepCollectionEquality();
 
+/// Determines an equality between [list1] and [list2].
```
Comment (felangel) [inline lib/src/equatable_utils.dart]:
```suggestion
/// Determines whether [list1] and [list2] are equal.
```

File: test/equatable_mixin_test.dart [hunk 6]
```diff
@@ -82,9 +82,9 @@ class Credentials extends EquatableBase {
   }
 
   Map<String, dynamic> toJson() {
-    final Map<String, dynamic> data = new Map<String, dynamic>();
-    data['username'] = this.username;
-    data['password'] = this.password;
+    final data = <String, dynamic>{};
```
Comment (felangel) [inline test/equatable_mixin_test.dart]: couldn't this just be
```dart
return <String, dynamic>{
  'username': username,
  'password': password,
};
```

Comment (tenhobi) [reply test/equatable_mixin_test.dart]: True, that would be even better. 😉

File: test/equatable_mixin_test.dart [hunk 7]
```diff
@@ -82,10 +82,10 @@ class Credentials extends EquatableBase {
...
 
   Map<String, dynamic> toJson() {
-    final data = <String, dynamic>{};
-    data['username'] = username;
-    data['password'] = password;
-    return data;
+    return {
+      'username': username,
+      'password': password,
+    };
```
Comment (tenhobi) [inline test/equatable_mixin_test.dart]: I have simplified it to this. Does this seem ok?

**Pull request merged**